prime-router/src/test/kotlin/fhirengine/azure/FHIRRouterIntegrationTests.kt

```diff
@@ -0,0 +1,552 @@
+                "digest": "${BlobAccess.digestToString(BlobAccess.sha256Digest(blobContents.toByteArray()))}",
+                "blobSubFolderName": "${sender.fullName}",
+                "topic": "${sender.topic.jsonVal}",
+                "schemaName": "${sender.schemaName}" 
+            }
+        """.trimIndent()
+    }
+
+    private fun getBlobContainerMetadata(): BlobAccess.BlobContainerMetadata {
+        val blobConnectionString =
+            """DefaultEndpointsProtocol=http;AccountName=devstoreaccount1;AccountKey=keydevstoreaccount1;BlobEndpoint=http://${azuriteContainer.host}:${
+                azuriteContainer.getMappedPort(
+                    10000
+                )
+            }/devstoreaccount1;QueueEndpoint=http://${azuriteContainer.host}:${
+                azuriteContainer.getMappedPort(
+                    10001
+                )
+            }/devstoreaccount1;"""
+        return BlobAccess.BlobContainerMetadata(
+            "container1",
+            blobConnectionString
+        )
+    }
+
+    private fun createReport(
+        fileFormat: Report.Format,
+        currentAction: TaskAction,
+        nextAction: TaskAction,
+        nextEventAction: Event.EventAction,
+        topic: Topic,
+        taskIndex: Long = 0,
+        childReport: Report? = null,
+        bodyURL: String? = null,
+    ): Report {
+        val report = Report(
+            fileFormat,
+            listOf(ClientSource(organization = universalPipelineOrganization.name, client = "Test Sender")),
+            1,
+            metadata = UnitTestUtils.simpleMetadata,
+            nextAction = nextAction,
+            topic = topic
+        )
+        ReportStreamTestDatabaseContainer.testDatabaseAccess.transact { txn ->
+            val action = Action().setActionName(currentAction)
+            val actionId = ReportStreamTestDatabaseContainer.testDatabaseAccess.insertAction(txn, action)
+            report.bodyURL = bodyURL ?: "http://${report.id}.${fileFormat.toString().lowercase()}"
+
+            val reportFile = ReportFile().setSchemaTopic(topic)
+                                         .setReportId(report.id)
+                                         .setActionId(actionId)
+                                         .setSchemaName("")
+                                         .setBodyFormat(fileFormat.toString())
+                                         .setItemCount(1)
+                                         .setExternalName("test-external-name")
+                                         .setBodyUrl(report.bodyURL)
+                                         .setSendingOrg(universalPipelineOrganization.name)
+                                         .setSendingOrgClient("Test Sender")
+
+            ReportStreamTestDatabaseContainer.testDatabaseAccess.insertReportFile(
+                reportFile, txn, action
+            )
+            if (childReport != null) {
+                ReportStreamTestDatabaseContainer.testDatabaseAccess
```

Comment:
what do you mean the item lineages are not being created?

Reply:
If you look at our database tables, you'll see report_lineage and item_lineage. We are updating the report_lineage table but ignoring the item_lineage table. The FHIRRouter does not depend on the item_lineage table to perform its function, but it does update it so that means we should mock it properly so we can verify the item_lineage is correct after the step has completed.

Reply:
I could see this being a different ticket so we can move forward with Routing refactor. @mkalish thoughts?